0000-automatic_features.md

```diff
@@ -0,0 +1,136 @@
+- Feature Name: `automatic_features`
+- Start Date: 2016-11-05
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Support "automatic" features, which will be used only if the dependencies of
+that feature are already present. Users will be able to write code to integrate
+with another library (for example, serializations for their type) without
+introducing a *necessary* dependency on that library for users who don't need
+it.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, a common pattern exists for libraries to have feature flags which
+turn on a dependency. For example, if I have a library which defines a new
+type, I might want users to be able to serialize it with `serde`, but I don't
+want users who don't care about this functionality to have to add `serde` to
+their dependency graph. So I add a `serialization` feature to my Cargo.toml,
+and only if you turn on this feature do you get the feature for serialization.
+
+Ultimately, the source of this problem is the orphan rules - because only *I*
+can define the impls for my type, I have to add this dependency. For every
+commonly used trait defining library, I have to add a new feature, and all of
+my clients have to turn on that feature.
+
+cargo features have many uses, but this particular pattern ends up being a lot
+of fiddling for a very common use case. Instead of requiring users go through
+the feature flag rigmarole, we can provide first class support for this
+pattern through "conditional dependencies" - dependencies that are turned on
+automatically if they were already present.
+
+# Detailed design
+[design]: #detailed-design
+
+## Automatic features in `Cargo.toml`
+
+When declaring a feature in a Cargo.toml file, if that feature is an
+object, it can contain the `automatic` member. For example:
+
+```toml
```

Comment:
The fact that (right now) you need to have a different name for the feature from the dependency is a big part of why I was in favor of the original proposal in the first place. Saying "automatic" could be applied to features is just returning to that proposal, but changing the key from "conditional" to "automatic" (which doesn't make sense to me).